Controllers/controller.py

```diff
@@ -106,3 +111,20 @@ def open_file_dialog(self):
             url = file_dialog.selectedUrls()[0]
             self._media_player.setSource(url)
             self._media_player.play()
+
+    @Slot()
+    def position_changed(self):
+        """Command the video player to change state of the position"""
+        position = self._media_player.position()
+        self._window.media_panel.progress_bar_slider.setValue(position)
+
+    @Slot()
+    def duration_changed(self):
+        """Command the video player to change the state when duration changed"""
+        duration = self._media_player.duration()
+        self._window.media_panel.progress_bar_slider.setRange(0, duration)
+
+    @Slot()
+    def set_position(self, position):
+        """Command the video player to set the state of position while dragging"""
```

Comment:
Let's be more specific about what's dragging:

"""
Commands the video player to set the position state based on the value of the progress bar slider.
"""
"""